crates/hir-ty/src/builder.rs

```diff
@@ -141,10 +144,13 @@ impl<D> TyBuilder<D> {
 
     #[tracing::instrument(skip_all)]
     pub(crate) fn fill_with_inference_vars(self, table: &mut InferenceTable<'_>) -> Self {
-        self.fill(|x| match x {
-            ParamKind::Type => table.new_type_var().cast(Interner),
-            ParamKind::Const(ty) => table.new_const_var(ty.clone()).cast(Interner),
-            ParamKind::Lifetime => table.new_lifetime_var().cast(Interner),
+        self.fill(|x| {
```

Comment:
There is a bunch of code that I migrated from `TyBuilder` to using methods on `GenericArgs` and the like; I regret this a bit, as `TyBuilder` is more principled and also checks against misuse, but it's too late now. We will want to consider what to do with the rest of the code that uses it though.